docs/docsite/rst/network/user_guide/cli_parsing.rst

```diff
@@ -0,0 +1,728 @@
+- ``textfsm`` was previously made available as a filter plugin. Ansible users should transition to the ``cli_parse`` module.
+
+.. note::
+
+	Red Hat Ansible Automation Platform subscription support is limited to the use of the ``textfsm`` public APIs as documented.
+
+This task sets the following fact as the ``interfaces`` fact for the host:
+
+.. code-block:: yaml
+
+   - ADDRESS: X254.005a.f8b5
+     ADMIN_STATE: up
+     BANDWIDTH: 1000000 Kbit
+     BIA: X254.005a.f8b5
+     DELAY: 10 usec
+     DESCRIPTION: ''
+     DUPLEX: full-duplex
+     ENCAPSULATION: ARPA
+     HARDWARE_TYPE: Ethernet
+     INPUT_ERRORS: ''
+     INPUT_PACKETS: ''
+     INTERFACE: mgmt0
+     IP_ADDRESS: 192.168.101.14/24
+     LAST_LINK_FLAPPED: ''
+     LINK_STATUS: up
+     MODE: ''
+     MTU: '1500'
+     OUTPUT_ERRORS: ''
+     OUTPUT_PACKETS: ''
+     SPEED: 1000 Mb/s
+   - ADDRESS: X254.005a.f8bd
+     ADMIN_STATE: up
+     BANDWIDTH: 1000000 Kbit
+     BIA: X254.005a.f8bd
+
+
+Parsing with TTP
+-----------------
+
+TTP is a Python library for semi-structured text parsing using
+templates. TTP uses a jinja-like syntax to limit the need for regular
+expressions. Users familiar with jinja templating may find the TTP template syntax familiar.
+
+The following is an example TTP template stored as ``templates/nxos_show_interfaces.ttp``:
+
+.. code-block:: jinja
+
+   {{ interface }} is {{ state }}
+   admin state is {{ admin_state }}{{ ignore(".\*") }}
+
+The following task uses this template to parse the ``show interface`` command output:
+
+.. code-block:: yaml
+
+   - name: "Run command and parse with ttp"
+     ansible.netcommon.cli_parse:
+       command: show interface
+       parser:
+         name: ansible.netcommon.ttp
+       set_fact: interfaces
+
+Taking a deeper dive in this task:
+
+- The default template path ``templates/nxos_show_interface.ttp`` was generated using the ``ansible_network_os`` for the host and ``command`` provided.
+- TTP supports several additional variables that will be passed to the parser. These include:
+
+  - ``parser/vars/ttp_init`` -  Additional parameter passed when the parser is initialized.
+  - ``parser/vars/ttp_results`` -  Additional parameters used to influence the parser output.
+  - ``parser/vars/ttp_vars`` -  Additional variables made available in the template.
+
+- See the `TTP documentation <https://ttp.readthedocs.io>`_ for details.
+
+
+The task sets the follow fact as the ``interfaces`` fact for the
+host:
+
+.. code-block:: yaml
+
+   - admin_state: up,
+     interface: mgmt0
+     state: up
+   - admin_state: up,
+     interface: Ethernet1/1
+     state: up
+   - admin_state: up,
+     interface: Ethernet1/2
+     state: up
+
+
+Converting XML
+-----------------
+
+Although Ansible contains a number of plugins that can convert XML to Ansible native data structures, the``cli_parse`` module runs the command on devices that return XML and returns the converted data in a single task.
```

Comment:
This is a great comparison - I might mention this higher up in the doc as well, since consolidating two tasks into one is likely a big selling point for this approach.